src/test/java/de/tobiasroeser/lambdatest/proxy/ExampleProxyTest.java

```diff
@@ -107,22 +119,105 @@ public boolean genericArg(String arg1) {
 				}
 			});
 			intercept(UnsupportedOperationException.class,
-					"(?s).*\\Qpublic boolean genericArg(Object object1)\\E.*",
+					"(?s).*\\Qpublic boolean genericArg(Object object0)\\E.*",
 					() -> dep.genericArg("1"));
 		});
-		test("A proxy with missing implementation should print a nice (copy 'n paste -able) method signature",
-				() -> {
-					final Dependency dep = TestProxy.proxy(Dependency.class);
-					intercept(UnsupportedOperationException.class,
-							"(?s).*\\Qpublic int foobar(String string1, int int1)\\E.*",
-							() -> dep.foobar("a", 1));
 
-					intercept(UnsupportedOperationException.class,
-							"(?s).*\\Qpublic Object baz(List list1)\\E.*",
-							() -> dep.baz(Arrays.asList("1")));
+		test("A proxy handling generics works with proposed code templates",() -> {
+			final Dependency<String> proxy = TestProxy.proxy(Dependency.class, new Object() {
+				// ==> public <T> T generic(T x0){}
+				public <T> T generic(T x0){ return x0; }
 
-				});
+				// ==> public U baz(List<U> x0){}
+				// works: public <X> X baz(List<X> x0){ return x0.get(0);}
+				// works: public String baz(List<String> x0){ return x0.get(0);}
+				// does not work public  int baz(List<Integer> x0){ return x0.get(0);}
+				public <X> X baz(List<X> x0){ return x0.get(0);}
 
+			});
+			Expect.expectString(proxy.generic("abc")).isEqual("abc");
+			Expect.expectString(proxy.baz(Arrays.asList("abc"))).contains("abc");
+		});
+
+		section("A proxy with with missing implementation should print a nice (copy 'n paste -able) method signature",
+				() -> {
+					test("case: int foobar(String s1, int i2)",
+							() -> {
+								final Dependency dep = TestProxy.proxy(Dependency.class, new Object() {
+									@SuppressWarnings("unused")
+									public String hello() {
+										return "Hello Proxy!";
+									}
+								});
```

Comment:
The delegate object isn't needed at all in this and also the next test cases.

```java
final Dependency dep = TestProxy.proxy(Dependency.class);
```

would be enough.

Reply:
done.